src/components/Badge/Badge.st.css

```diff
@@ -0,0 +1,316 @@
+:import {
+  -st-from: "wix-ui-core/dist/src/components/StylableBadge/BadgeStyle.st.css";
+  -st-default: Badge;
+}
+
+:import {
+  -st-from: "wix-ui-core/dist/src/components/StylableBadge/Badge.st.css";
+  -st-default: BaseBadge;
+}
+
+:import {
+  -st-from: "../../palette.st.css";
+  -st-named: secondaryText, primaryLightText, mainHover, danger, success, placeholderText, warning, urgent, mainMuted, successMutedHover, backgroundSecondary, dangerMutedHover;
+}
+
+.root {
+  -st-extends: BaseBadge;
+  -st-states: skin(string), type(string);
+}
+
+.root >:not(:last-child) {
+  margin-right: 8px;
+}
+
+.prefix, .suffix {
+  width: 12px;
+  height: 12px;
+}
+
+/* children color when type = solid */
+.root:type(solid):skin(default) .text,
```

Comment:
I think that you suggested targeting the `UIText` instead of passing `.text` classname.

```css
:import {
  -st-from: "../StylableUIText/UIText.st.css";
  -st-default: UIText;
}

.root:type(solid):skin(default) UIText {
  ...
}
```

Reply:
This was my initial implementation, but then when I had an issue int the tests then I sat with Barak and he advised that having a className is better as it exposes the stylesheet api.

Reply:
ok